### DISCUSSION TITLE
Altering: being able to control the exact cycle of the execution window

### DISCUSSION THREAD

**Discussion by robinwangrubin** [grafana/grafana#48216] [Alerting]
**What happened**:
My alert will never be touched or always be touched. when i used percent diff function to monitor the increate  rate.

**What you expected to happen**:

percent_diff() or diff() should always run in integer multiples:  

example in my case; the Evaluate should be run at :  10:00    10:05    10:10   10:15
![image](https://user-images.githubusercontent.com/24934186/164365867-6f095471-dd37-4cfe-9f45-4c76c674fc01.png)

the Data should be like this:  
```
[0:Array[121747,1650510000000]]
[1:Array[80598,1650510300000]]
```

Real situation; the Evaluate runs at :             10:02    10:07    10:12   10:17
what i got is like this:
```
[0:Array[105915,1650510075177]]
[1:Array[80598,1650510300000]]
[2:Array[13494,1650510600000]]
```

a few minutes later:
```
[0:Array[98825,1650510152902]]
[1:Array[80598,1650510300000]]
[2:Array[50681,1650510600000]]
```

Because it gets the first and last elements of the data array and diffs them. in this case, if the Evaluate run at 10:00 10:05 10:10, then everything would be coreect. But when Evaluate run at 10:02  10:07  10:12  10:17, it will always get 3 group data. first group data is not complete. medial group is fine. and final group is not complete. because i used window(every: 5m).  then the diff resualt is wrong.

**How to reproduce it (as minimally and precisely as possible)**:
```
from(bucket: "BlockChain") 
    |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
    |> filter(fn: (r) => r._measurement == "Binance" and r._field == "Quantity" and r.Symbol=="LRCUSDT")
    |> group()
    |> sort(columns: ["_time"])
    |> window(every: 5m)
    |> sum()
    |> duplicate(column: "_start", as: "_time")
    |> window(every: inf) 
```

**Anything else we need to know?**:

**Environment**:
- Grafana version: Grafana v8.2.5 (b57a137acd)
- Data source type & version: Influxdb 2.2
- OS Grafana is installed on: Centos 7.8
- User OS & Browser: Mac Chrome
- Grafana plugins: None
- Others:

**Comment by JohnnyQQQQ** [comment]
Hey 👋  thanks for raising this issue. Currently, it is not possible to control the exact execution cycle of a window. I'll rephrase the issue and convert this to a feature request.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
